src/main/java/DungeonController.java

```diff
@@ -0,0 +1,242 @@
+                    addGold(r, startHeight, startWidth);
+                    break;
+
+                case 4:
+                    startHeight = rand.nextInt((dungeon.getHeight() - 7) - (dungeon.getHeight() / 2 + 2)) + (dungeon.getHeight() / 2 + 2);
+                    startWidth = rand.nextInt((dungeon.getWidth() - 7) - (dungeon.getWidth() / 2 + 2)) + (dungeon.getWidth() / 2 + 2);
+                    addGold(r, startHeight, startWidth);
+                    break;
+
+                default:
+                    startHeight = rand.nextInt(1);
+                    startWidth = rand.nextInt(1);
+            }
+
+
+            for (int i = 0; i < 5; i++) {
+                //draw room (y, x)
+                dungeonMatrix[i + startHeight][startWidth] = "*"; //left
+                dungeonMatrix[startHeight][i + startWidth] = "*"; //top
+                dungeonMatrix[i + startHeight][startWidth + 4] = "*"; //right
+                dungeonMatrix[startHeight + 4][i + startWidth] = "*"; //bottom
+
+                gridBounds[i + startHeight][startWidth] = 1; //left
+                gridBounds[startHeight][i + startWidth] = 1; //top
+                gridBounds[i + startHeight][startWidth + 4] = 1; //right
+                gridBounds[startHeight + 4][i + startWidth] = 1; //bottom
+
+            }
+            removeDoor(startHeight, startWidth);
+        }
+    }
+
+    private void addGold(int room, int sh, int sw) {
+        golds[room - 1] = new Gold();
+        golds[room - 1].setGoldX(sw + 2);
+        golds[room - 1].setGoldY(sh + 2);
+
+        dungeonMatrix[sh + 2][sw + 2] = golds[room - 1].getGoldSymbol();
+        gridBounds[sh + 2][sw + 2] = 2;
+    }
+
+    private void removeDoor(int sh, int sw) {
+        Random rand = new Random();
+        int wall = rand.nextInt(4);
+
+        switch (wall) {
+            case 0: //Top wall
+                dungeonMatrix[sh][sw + 2] = " ";
+                gridBounds[sh][sw + 2] = 0;
+                break;
+            case 1: //Bottom Wall
+                dungeonMatrix[sh + 4][sw + 2] = " ";
+                gridBounds[sh + 4][sw + 2] = 0;
+                break;
+            case 2: //Right Wall
+                dungeonMatrix[sh + 2][sw + 4] = " ";
+                gridBounds[sh + 2][sw + 4] = 0;
+                break;
+            case 3: //Left Wall
+                dungeonMatrix[sh + 2][sw] = " ";
+                gridBounds[sh + 2][sw] = 0;
+                break;
+        }
+    }
+
+    public void addScore() {
+        if (gridBounds[player.getPlayerY()][player.getPlayerX()] == 2) {
+            gridBounds[player.getPlayerY()][player.getPlayerX()] = 0;
+            player.setGold(player.getGold() + golds[0].getGoldQuantity());
+        }
+    }
+
+    public void checkExit() {
+        if (gridBounds[player.getPlayerY()][player.getPlayerX()] == 5) {
+            if (player.getGold() >= gameWinAmount) {
+                System.out.println("Completed");
```

Comment:
Is this debug, if not the user probably wont see this unless they launch from a shell